pkg/mcptools/topology.go

```diff
@@ -0,0 +1,98 @@
+package mcptools
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/redhat-appstudio/tssc-cli/pkg/chartfs"
+	"github.com/redhat-appstudio/tssc-cli/pkg/config"
+	"github.com/redhat-appstudio/tssc-cli/pkg/constants"
+	"github.com/redhat-appstudio/tssc-cli/pkg/resolver"
+
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mark3labs/mcp-go/server"
+)
+
+// TopologyTool represents the MCP tool that's responsible to report
+// the dependency topology of the installer based on the cluster
+// configuration and Helm charts.
+type TopologyTool struct {
+	cfs *chartfs.ChartFS
+	cm  *config.ConfigMapManager
+}
+
+const (
+	// TopologyToolName mcp topology tool name
+	TopologyToolName = constants.AppName + "_topology"
+)
+
+// topologyHandler shows a table of the topology.
+func (t *TopologyTool) topologyHandler(
+	ctx context.Context,
+	_ mcp.CallToolRequest,
+) (*mcp.CallToolResult, error) {
+	// Load all charts from the embedded filesystem, or from a local directory.
+	charts, err := t.cfs.GetAllCharts()
+	if err != nil {
+		return nil, err
+	}
+	// Create a new chart collection from the loaded charts.
+	collection, err := resolver.NewCollection(charts)
+	if err != nil {
+		return nil, err
+	}
+	// Load the installer configuration from the cluster.
+	cfg, err := t.cm.GetConfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// Resolving the dependency topology based on the installer configuration and
+	// Helm charts.
+	r := resolver.NewResolver(cfg, collection, resolver.NewTopology())
+	if err := r.Resolve(); err != nil {
+		return nil, err
+	}
```

Comment:
Please use the [TopologyBuilder](https://github.com/redhat-appstudio/tssc-cli/blob/e6e5a1fca1dc2d7851b4e2a2bde028f25b7ed829/pkg/resolver/topologybuilder.go), which is already instantiated in a practical way to share it with your `TopologyTool`:

https://github.com/redhat-appstudio/tssc-cli/blob/e6e5a1fca1dc2d7851b4e2a2bde028f25b7ed829/pkg/subcmd/mcpserver.go#L70-L73